sale_deposit_deduction_option/wizard/sale_make_invoice_advance.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2023 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class SaleAdvancePaymentInv(models.TransientModel):
+    _inherit = "sale.advance.payment.inv"
+
+    deposit_deduction_option = fields.Selection(
+        [
+            ("full", "Deduct Full Deposit"),
+            ("proportional", "Deduct Deposit Proportionally"),
+            ("no", "No Deduct"),
+        ],
+        default="full",
+        required=True,
+    )
+
+    @api.onchange("deposit_deduction_option")
+    def _onchange_deposit_deduction_option(self):
+        self.deduct_down_payments = self.deposit_deduction_option != "no"
+
+    def create_invoices(self):
+        if self.deposit_deduction_option == "proportional":
+            sale_orders = self.env["sale.order"].browse(
+                self._context.get("active_ids", [])
+            )
+            self = self.with_context(
```

Comment:
@Saran440 I don't like passing things to context if that can be avoided.

Here, you can override _create_invoice() as it returns the created move. Then, you can do the process you have written in account.move.create().

Reply:
@Saran440 Any news on this ?

Reply:
@rousseldenis Hi, sorry for late reply.

Because user can create invoice either proportional or not each time. (Wizard view)
I don't know how to do if no passed value in context.
it can overwrite function, but this module maybe can't use with other modules.